### PR TITLE
Added missing documentation about TcpListener::listen and Socket::AnyPort

### PR DESCRIPTION
**Pull request by Unarelith**
## Description

As the title says.

This PR is related to the issue #1657 

## Tasks

* [x] Tested on Linux
* [x] Tested on Windows
* [ ] Tested on macOS
* [ ] Tested on iOS
* [ ] Tested on Android

### CODE REVIEW

File: include/SFML/Network/TcpListener.hpp [hunk 1]
```diff
@@ -74,6 +74,10 @@ class SFML_NETWORK_API TcpListener : public Socket
     /// function is called, it will stop listening on the old
     /// port before starting to listen on the new port.
     ///
+    /// You can use the special value Socket::AnyPort to tell the
```
Comment (eXpl0it3r) [inline include/SFML/Network/TcpListener.hpp:77]: Can you use the full qualifying name `sf::Socket::AnyPort`

File: include/SFML/Network/TcpListener.hpp [hunk 2]
```diff
@@ -74,6 +74,10 @@ class SFML_NETWORK_API TcpListener : public Socket
...
+    /// system to automatically pick an available port, and then
+    /// call getLocalPort to retrieve the chosen port.
```
Comment (eXpl0it3r) [inline include/SFML/Network/TcpListener.hpp:79]: I'd have probably gone more with something like this:

```
/// When providing sf::Socket::AnyPort as port, the listener
/// will request an available port from the system.
/// The chosen port can be retrieved by calling getLocalPort().
```

Comment (Unarelith) [reply include/SFML/Network/TcpListener.hpp:79]: I basically copy-pasted the lines from UdpSocket.hpp: https://github.com/SFML/SFML/blob/master/include/SFML/Network/UdpSocket.hpp#L81-L83

Comment (Unarelith) [reply include/SFML/Network/TcpListener.hpp:79]: But I prefer your way. Do you want me do edit UdpSocket.hpp too to keep consistency?

Comment (eXpl0it3r) [reply include/SFML/Network/TcpListener.hpp:79]: Oh I didn't know. Still kind of prefer my version. Either way, I think it makes sense to have the `sf::` prefix in both versions.

Comment (Unarelith) [reply include/SFML/Network/TcpListener.hpp]: Ok done. :)

**Comment by Unarelith** [comment]
Sure

**Pull request merged**